lib/push.js

```diff
@@ -35,25 +44,31 @@ module.exports = function (log, db) {
             if (device.pushCallback) {
               // send the push notification
               log.increment(LOG_UPDATE_INCREMENT_SEND)
-              request.post(device.pushCallback, function (err){
+              request.post(device.pushCallback, function (err, response) {
                 if (err) {
-                  log.error({
-                    op: LOG_OP_NOTIFY_UPDATE,
-                    deviceId: deviceId,
-                    err: err
-                  })
-                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                  // any 400 level error from the push servers means
+                  // the push settings need to be reset.
+                  // the clients will check this and re-register push endpoints
+                  if (response && response.statusCode >= 400 && response.statusCode < 500) {
+                    // reset device push configuration
+                    // Warning: this method is called without any session tokens or auth validation.
+                    device.pushCallback = ''
+                    device.pushPublicKey = ''
```

Comment:
@philbooth should this be an empty string or an empty `Buffer` object? I noticed in the tests it is a Buffer, but `null` by default.